packages/evm/contracts/payments/PaymentsBook.sol

```diff
@@ -0,0 +1,535 @@
+    */
+    function getCurrentPeriodId() external view returns (uint256) {
+        return _getCurrentPeriodId();
+    }
+
+    /**
+    * @dev Get the fee details of a payment period
+    * @param _periodId Identification number of the period to be queried
+    * @param _token Address of the token querying the fee details for
+    * @return jurorFees Juror fees for the requested period and token
+    * @return governorFees Governor fees for the requested period and token
+    */
+    function getPeriodFees(uint256 _periodId, address _token)
+        external
+        view
+        returns (uint256 jurorFees, uint256 governorFees)
+    {
+        Period storage period = periods[_periodId];
+        jurorFees = period.jurorFees[_token];
+        governorFees = period.governorFees[_token];
+    }
+
+    /**
+    * @dev Get the balance details of a payment period
+    * @param _periodId Identification number of the period to be queried
+    * @return balanceCheckpoint Court term ID of a period used to fetch the total active balance of the jurors registry
+    * @return totalActiveBalance Total amount of juror tokens active in the Court at the corresponding period checkpoint
+    */
+    function getPeriodBalanceDetails(uint256 _periodId)
+        external
+        view
+        returns (uint64 balanceCheckpoint, uint256 totalActiveBalance)
+    {
+        Period storage period = periods[_periodId];
+        balanceCheckpoint = period.balanceCheckpoint;
+        totalActiveBalance = period.totalActiveBalance;
+    }
+
+    /**
+    * @dev Tell the fees corresponding to a juror for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _juror Address of the juror querying the owed fees of
+    * @param _token Address of the token to be queried
+    * @return Token amount corresponding to the juror
+    */
+    function getJurorFees(uint256 _periodId, address _juror, address _token) external view returns (uint256) {
+        require(_periodId < _getCurrentPeriodId(), ERROR_NON_PAST_PERIOD);
+
+        Period storage period = periods[_periodId];
+        uint256 totalActiveBalance = period.totalActiveBalance;
+        require(totalActiveBalance != 0, ERROR_PERIOD_BALANCE_DETAILS_NOT_COMPUTED);
+
+        uint256 jurorActiveBalance = _getJurorActiveBalance(_juror, period.balanceCheckpoint);
+        return _getJurorFees(period, _token, jurorActiveBalance, totalActiveBalance);
+    }
+
+    /**
+    * @dev Tell the fees corresponding to a juror for a certain period
+    * @param _periodId Identification number of the period being queried
+    * @param _juror Address of the juror querying the owed fees of
+    * @param _tokens List of token addresses to be queried
+    * @return List of token amounts corresponding to the juror
+    */
+    function getManyJurorFees(uint256 _periodId, address _juror, address[] calldata _tokens) external view returns (uint256[] memory amounts) {
```

Comment:
I'm not against these "many" variants, but similar to before, I'm curious how much we want them.

On one side, it's a slight UX gain / gas gain if we anticipate another on-chain contract to call the version with just a single token. On the other hand, we don't necessarily need the "many" version for off-chain interactions, as one could do a multicall (or just multiple `eth_call`s).

At the moment, similar to before, I'm leaning towards simplifying to only keeping the "many" version. Are there any potential problems with subgraph logic if we do this, or is it already just based on the `PaymentReceived` events?

Reply:
I agree, addressed in 51b7ed3c9324e62bd2b9bc9b85047db0c062ac7b